src/App.js

```diff
@@ -7,66 +7,105 @@ import {
   NavLink,
   Redirect,
 } from 'react-router-dom';
+import { db } from './lib/firebase';
+// import { useCollection } from 'react-firebase-hooks/firestore';
+import { SnackbarProvider } from 'notistack';
 
 import AddItems from './AddItems';
 import ItemList from './ItemList';
 import Welcome from './Welcome';
 
-
 function App() {
-  const hasToken = localStorage.getItem('userToken');
+  const [token, setToken] = useState(localStorage.getItem('userToken'));
+  const [list, setList] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    let unsubscribe;
+    try {
+      unsubscribe = db.collection(token).onSnapshot((snapshot) => {
+        const newList = [];
+        snapshot.forEach((doc) => {
+          // This seems to work, while snapshot.map doesn't
+          newList.push(
+            JSON.stringify(doc.data()['formData']['itemName']).replace(
+              /['"]+/g,
+              '',
+            ),
+          );
+        });
+        setList(newList);
+        setError(null);
+        setLoading(false);
+      });
+    } catch (error) {
+      setError("Can't connect to the database");
+      setLoading(false);
+    }
+    return () => {
+      unsubscribe();
+    };
```

Comment:
Nice! Is this return function in order to prevent memory leaks? I'm curious because it seems to be working even when I comment out this clean-up function.

Reply:
Yes, that's exactly what it does. The return statement of `useEffect` removes the listener that we created in this variable. If we don't clean it up, the app will still work, but the listener will keep working even after the component unmounts. So as `App` re-renders, we may end up with multiple listeners. At least that's my understanding 😄

Reply:
That's awesome! Yea that totally makes sense to me!

Reply:
Nice job with this! It's great practice to do this whenever you subscribe to something that might not always be there, like event listeners and other active watchers like Firestore.

Reply:
I just wanted to note that larger projects have a very similar approach for error handling for API requests, however to prevent repetitive boiler plate code it is usually kept inside of a custom hook that can be called from anywhere. If you find yourself wanting to using similar error checking in other parts of our app, I'd definitely recommend bundling it together in a hook to keep everything clean and DRY. Nice work setting this up!

Reply:
Oh wow, I definitely had to look up how unsubscribe works in React. Daria, your explanation helped me even more - thank you!

Reply:
Thanks, @luisaugusto! We'll make sure to put this into a custom hook if we want to reuse it down the road.